libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1702,21 +1702,27 @@ std::optional<std::pair<PeriodData, std::vector<std::shared_ptr<Vote>>>> PbftMan
   }
   auto non_finalized_transactions = trx_mgr_->excludeFinalizedTransactions(transactions_to_query);
 
-  if (non_finalized_transactions.size() != period_data.transactions.size()) {
-    LOG(log_er_) << "Synced PBFT block " << pbft_block_hash << " transactions count " << period_data.transactions.size()
-                 << " incorrect, expected: " << non_finalized_transactions.size();
-    sync_queue_.clear();
-    net->handleMaliciousSyncPeer(node_id);
-    return std::nullopt;
-  }
-  for (uint32_t i = 0; i < period_data.transactions.size(); i++) {
-    if (!non_finalized_transactions.contains(period_data.transactions[i]->getHash())) {
-      LOG(log_er_) << "Synced PBFT block " << pbft_block_hash << " has incorrect transaction "
-                   << period_data.transactions[i]->getHash();
+  const uint32_t sanity_check_interval = 100;
+  // Next section should not be possible for a valid cert voted block, perform check only each sanity_check_interval not
+  // to hurt performance of syncing/rebuilding
+  if (period_data.pbft_blk->getPeriod() % sanity_check_interval == 0) {
```

Comment:
Is it taking a lot of time? 
I think it's better to have it every block or don't have at all. Because if it is fine on 100 block it doesn't mean that it is ok for previous blocks

Reply:
It is taking about 1.8millisecond per block. So it is 1.8s per 1000 block or if we look at mainnet with 5m blocks it is about 3 hours of processing when processing full mainnet blocks.

Reply:
I removed it completely